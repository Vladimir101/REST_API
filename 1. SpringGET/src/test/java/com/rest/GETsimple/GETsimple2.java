package com.rest.GETsimple;

import org.approvaltests.Approvals;
import org.json.JSONException;
import org.json.JSONObject;
import org.junit.jupiter.api.*;
import org.springframework.web.client.RestTemplate;

import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.rest.GETsimple.pages.PostCode;

class GETsimple2
{
	private RestTemplate restTemplate;
	private String URL = "http://api.zippopotam.us/us/94404";

	@BeforeEach
	void setUp() throws Exception
	{
		restTemplate = new RestTemplate();
	}

	@Test
	void test1() throws JsonProcessingException
	{
		PostCode response = restTemplate.getForObject(URL, PostCode.class);
		ObjectMapper mapper = new ObjectMapper();
		String json = mapper.writerWithDefaultPrettyPrinter().writeValueAsString(response);
		System.out.println(response.getPost_code());
		System.out.println(response.getCountry_abbreviation());
		System.out.println(response.getPlaces().get(0).getPlace_name());
		Approvals.verify(json);
	}
	

	@Test
	void test2() throws JSONException
	{
		String response = restTemplate.getForObject(URL, String.class);
		JSONObject json = new JSONObject(response); // Convert text to object
		System.out.println("Pretty print:");
		System.out.println(json.toString(4)); // Print it with specified indentation
		Approvals.verify(response);
	}
}
