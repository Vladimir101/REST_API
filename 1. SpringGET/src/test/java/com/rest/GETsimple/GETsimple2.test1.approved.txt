{
  "country" : "United States",
  "places" : [ {
    "longitude" : "-122.27",
    "state" : "California",
    "latitude" : "37.5538",
    "place name" : "San Mateo",
    "state abbreviation" : "CA"
  } ],
  "post code" : "94404",
  "country abbreviation" : "US"
}